### ISSUE TITLE
[no-invalid-void-type] another use case for void type

### ISSUE THREAD

**Issue by yangmingshan** [typescript-eslint/typescript-eslint#2165]
**Repro**

https://www.typescriptlang.org/play/index.html#code/PTAEBUAsEsGdQO7QDbNAYwIYFdYFNQ8AnIgeyICgAzbAO3QBdpTbQG9YGAKBmWALlB0AJnirRaeYQEpQAb1ABfChRCgAQtgZs+iFGgBGBcZNVga9JizYduvOIIBupaDPlKV7Tl2lA

```JSON
{
  "rules": {
    "@typescript-eslint/no-invalid-void-type": "error"
  }
}
```

```TS
function test(this: void) { }
test()
```

**Expected Result**
No error.

**Actual Result**
Error: void is only valid as a return type or generic type variable

**Additional Info**
I think this is another valid use case that should be supported. Real life use case: https://github.com/vuejs/vue-next/pull/605

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.1.0` |
| `@typescript-eslint/parser`        | `3.1.0` |
| `TypeScript`                       | `3.9.3` |
| `ESLint`                           | `7.1.0` |
| `node`                             | `14.4.0` |
| `npm`                              | `6.14.5` |

**Comment by bradzacher** [comment]
See #2148
Please use the search
